### ISSUE TITLE
[meta] Migrating inline `TODO` comments to GitHub issues

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1004]
https://github.com/textpattern/textpattern/search?utf8=✓&q=todo&type=

There are currently 24 instances of `TODO` in the production code. These should be extracted and migrated to proper issues. Immediate result with make the production code marginally smaller, and increase visibility of things that need attention. Am I missing any good reason why this shouldn't happen?

**Comment by philwareham** [comment]
The TODOs in my CSS (if any) are notes to self generally. But I guess it would be good to at least audit the TODOs in the system to see if any are obsolete/immediately solvable.

**Comment by Bloke** [comment]
Good idea. During extraction, we should make a note of where in the file(s) a ToDo was mentioned (bearing in mind that line numbers change over time) because some of them are very specific to particular sections of code. That's usually why they've remained inline rather than becoming issues in their own right, because they're specific to one or two lines of code or, as Phil says, memory joggers.

But there are entries that definitely should be raised here. Some can even be amalgamated, e.g. all the ones about removing CF stuff from txp_prefs will be covered in the Custom Fields improvements.

Basically, I'd keep any little inline notes that are specific to one or two lines of code, and extract the rest that are bigger in scope.

Oh, and I wouldn't worry about the RPC ones unless we find some overarching reason to worry about it again.

**Comment by petecooper** [comment]
OK, thanks both. Proposal:

* grep the code (just `textpattern/textpattern`) for TODOs
* open issue with the TODO comment, plus surrounding code blocks
* if it's valid, update the TODO inline with the GH issue reference
* if not valid, scrub the comment and close the issue

That keeps the code clean and relevant, and there's an audit trail for any future picking through.

I can take this as a task as I realise you're all working toward 4.7 release stuff.

**Comment by Bloke** [comment]
Sounds like a plan, thanks for picking this up.

**Comment by petecooper** [comment]
Closing for now, will open new issues as required.

**Comment by petecooper** [comment]
OK, I'll do these in small batches so they're not overwhelming.

**Comment by makss** [comment]
I will now be afraid to write the words TODO and FIXME(Oops, why did I say that? ;-)  )

**Comment by petecooper** [comment]
@makss - do not be afraid!

I started this issue because as a non-developer, I watch more people involved with developing, designing, testing and helping out with Textpattern than I have in >10 years. There is momentum, which is excellent.

Code TODOs and FIXMEs are an extra level hidden from GitHub issues. Some of the TODOs are from a previous era and may not be relevant today. Please consider it housekeeping. If someone wishes to become involved and observes an issue which they can help with, along with notes and background information, then everyone will benefit.

So, please do not be afraid.

(Well, OK, maybe a little…)

**Comment by petecooper** [comment]
I've given this some extended thought -- I'm not sure it's useful with the amount and scope of changes going on. I'll close this for now, along with the open issues that have no work-in-progress actions. Perhaps revisit after 4.7 drops.
